### PR TITLE
[ShaderGraph] [VT] Fixes to VT and package compatibility with DOTS [Skip CI]

### PR DESCRIPTION
**Pull request by cdxntchou**
### Purpose of this PR
Fixes a number of issues with ShaderGraph and the Virtual Texture integration:
* Handle DOTS style assemblies (that throw exceptions when you call GetTypes() on them)
* Sample Virtual Texture node was missing the AddressMode control
* Sample Virtual Texture "No Feedback" toggle changed to "Automatic Streaming"
* Sample Virtual Texture generated HLSL function now has a better name (no collision issues)

---
### Testing status

Opened a number of VT use cases in editor (windows, DX11), including converted Environment projects and local VT test project.  Ran through about half of the VT behavior test checklist.

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by LandonTownsendUnity** [comment]
Aside from that, I added the Hybrid Renderer to the project and there were no compile errors, so that's validated.

**Comment by cdxntchou** [comment]
> 
> 
> "Udim" is not supported, you should hide that option in Address Mode

Removed it

**Comment by cdxntchou** [comment]
> 
> 
> If you have a virtual texture property and sample node that is plugged into the master stack and is compiling (all textures are assigned) and then you create a new virtual texture property, it throws this error and softlocks the graph:

Verified this is caused by duplicate layer names.  Liz's PR that is already merged to master fixes it by making the default Layer names more unique.

File: com.unity.shadergraph/Editor/Drawing/Inspector/PropertyDrawers/SampleVirtualTextureNodePropertyDrawer.cs [hunk 1]
```diff
@@ -52,14 +65,14 @@ public class SampleVirtualTextureNodePropertyDrawer : IPropertyDrawer
...
+                    if (node.noFeedback == !newValue)
                         return;
 
                     node.owner.owner.RegisterCompleteObjectUndo("Feedback Settings Change");
-                    node.noFeedback = newValue;
+                    node.noFeedback = !newValue;
                 },
-                node.noFeedback,
-                "No Feedback",
+                !node.noFeedback,
```
Comment (elizabeth-legros) [inline com.unity.shadergraph/Editor/Drawing/Inspector/PropertyDrawers/SampleVirtualTextureNodePropertyDrawer.cs:74]: Would prefer this be more explicit in what its doing

**Comment by marctem** [comment]
All CI failures match current master run.

**Pull request merged**